### PR TITLE
Blocks: Reselect the current block if we click again on it

### LINKED ISSUE #498: Unclear how to enable controls after typing
Once you type in an editable field, the controls disappear, and it is unclear how to make them show again. The only way seems to be to select another block, and then select then old one again.

Maybe we should always show controls on `mousedown touchstart`?

### PR DESCRIPTION
**Pull request by youknowriad**
closes #498 

This PR allows reselecting a block (showing the controls) when clicking on the block again.

**Testing instructions**

 - Select a block
 - Type some text on it
 - Click on the block again
 - The controls should show up

### CODE REVIEW

**Comment by iseulde** [comment]
Works really well. I'm experiencing a strange bug where, if I click on a block to show the controls, and then click on another, the caret appears on the start instead of the letter clicked.

**Comment by mtias** [comment]
I was seeing a similar thing but triggered by clicking on one of the formatting buttons repeatedly (three times on italics without pressing any other key).

**Comment by youknowriad** [comment]
@mtias @iseulde These issues should be fixed by 473563a

The idea is to avoid unnecessary `updateContent` calls.

File: editor/modes/visual-editor/block.js [hunk 1]
```diff
@@ -137,15 +137,13 @@ class VisualEditorBlock extends wp.element.Component {
 			wrapperProps = settings.getEditWrapperProps( block.attributes );
 		}
 
-		// Disable reason: Each block can receive focus but must be able to contain
-		// block children. Tab keyboard navigation enabled by tabIndex assignment.
+		// Disable reason: Each block can be selected by clicking on it
```
Comment (aduth) [inline editor/modes/visual-editor/block.js]: The changes here break tab navigation:

1. Focus text block
2. Press tab

Expected: Image block is selected
Actual: Third block (next text block) is selected

The disable reason doesn't seem acceptable for accessibility.

File: blocks/components/editable/index.js [hunk 2]
```diff
@@ -266,10 +268,13 @@ export default class Editable extends wp.element.Component {
 			this.focus();
 		}
 
+		// The savedContent var allows us to avoid updating the content right after an onChange call
```
Comment (aduth) [inline blocks/components/editable/index.js]: I'm not sure I follow this. Why does the value of `this.props.value` differ from `prevProps.value` after the `onChange`? And if it does, why do we not want this to be reflected in the editable?

Comment (youknowriad) [inline blocks/components/editable/index.js]: if you make a change on the editable. Say the old value is `a` and the new one is `ab`, we don't want to trigger the `updateContent` even if the old value prop and the new one are different. That's what I'm trying to achieve here by caching the  `savedContent`.

File: editor/modes/visual-editor/block.js (hunk 1 above)
Comment (youknowriad) [inline editor/modes/visual-editor/block.js]: Fixed in 11d81e4

Comment (aduth) [inline editor/modes/visual-editor/block.js]: Should the comment be reverted back to its previous text? Or at least updated to reflect that it accepts focus? Seems like suggesting "clicking on it" as reason for disabling an accessibility rule is destined to... provoke response 😄

**Review by aduth — APPROVED** [review]

**Pull request merged**